otter/convergence/gathering.py

```diff
@@ -211,9 +210,9 @@ def update_drained_at(node):
         if node.description.lb_id in deleted_lbs:
             return None
         if feed is not None:
-            return assoc_obj(node, drained_at=extract_clb_drained_at(feed))
-        else:
-            return node
+            node.drained_at = extract_clb_drained_at(feed)
```

Comment:
As far as calling `assoc` with a private variable... that seems like a bug in `attrs` which this is just a workaround for (and you're not reaching across library boundaries, at least)